File: src/main/java/com/home/demos/samplebankmodule/model/Payment.java
package com.home.demos.samplebankmodule.model;

import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration;
import lombok.Data;

import java.util.List;

@Data
public class Payment {
    private Long id;
    private String title;
    private Long clientId;
    private String target;
    private Long sum;

    public Payment(Long clientId) {
        this.clientId = clientId;
    }

    public Payment() {
    }

    public Payment create() {
        return SampleBankModuleConfiguration
                .paymentRepository()
                .save(this);
    }

    public List<Payment> findAllLikeThis() {
        return SampleBankModuleConfiguration
                .paymentRepository()
                .findAllByClientId(this.clientId);
    }
}

File: src/main/kotlin/com/home/demos/samplebankmodule/model/Payment.kt
package com.home.demos.samplebankmodule.model

import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration.Companion.asyncRepository
import com.home.demos.samplebankmodule.config.SampleBankModuleConfiguration.Companion.paymentRepository

class Payment(
        val id: Long,
        val title: String,
        var clientId: Long,
        val target: String,
        val sum: Long
) {

    fun create(): TransactionId {
        return asyncRepository().process(this) { paymentRepository().save(this) }
    }

    fun findAllLikeThis(): TransactionId {
        return asyncRepository().process(this) { paymentRepository().findAllByClientId(clientId) }
    }
}
